### PR TITLE
[optimize](parquet-reader) Skip whole row group in the parquet lazy read situation if data has been filtered out.

### LINKED ISSUE #19038: [Enhancement] qt_q11 in regression test test_external_catalog_hive is very slow.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

We found `qt_q11` in regression test `test_external_catalog_hive` is very slow. 
The result is only one record, so other data should be filtered out in the parquet lazy read situation.
Then we found currently the parquet reader read many records because we can only skip parquet page. But in order to skip parquet page, currently we need to read page header, then it will caused prefetch data. Therefore, prefetch data in this case may be not good.

So there are two issues:

1. Skip whole row group in this case.
2. Prefetching data in this case may be not good, need to improve it.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by kaka11chen**
# Proposed changes

## Problem summary

Close #19038

We found `qt_q11` in regression test `test_external_catalog_hive` is very slow. 
The result is only one record, so other data should be filtered out in the parquet lazy read situation.
Then we found currently the parquet reader read many records because we can only skip parquet page. But in order to skip parquet page, currently we need to read page header, then it will caused prefetch data. Therefore, prefetch data in this case may be not good.

So there are two issues:

1. Skip whole row group in this case.
2. Prefetching data in this case may be not good, need to improve it.

This PR resolve issues 1.
 
### Test result:
Before opt:
```
mysql> select l_quantity from test_external_catalog_hive.tpch_1000_parquet.lineitem where l_orderkey = 599614241 and l_partkey = 59018738 and l_suppkey = 1518744 limit 2;
+------------+
| l_quantity |
+------------+
|      16.00 |
+------------+
1 row in set (2 min 27.55 sec)
```

After opt:
```
mysql> select l_quantity from test_external_catalog_hive.tpch_1000_parquet.lineitem where l_orderkey = 599614241 and l_partkey = 59018738 and l_suppkey = 1518744 limit 2;
+------------+
| l_quantity |
+------------+
|      16.00 |
+------------+
1 row in set (41.95 sec)
```

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp [hunk 1]
```diff
@@ -450,6 +448,17 @@ Status RowGroupReader::_do_lazy_read(Block* block, size_t batch_size, size_t* re
...
             Block::erase_useless_column(block, origin_column_num);
+
+            if (!pre_eof) {
+                // If continuous batches are skipped, we can cache them to skip a whole page
+                _cached_filtered_rows += pre_read_rows;
+            } else { // pre_eof
+                // If select_vector_ptr->filter_all() and pre_eof, we can skip whole row group.
+                *read_rows = 0;
+                *batch_eof = true;
+                _lazy_read_filtered_rows += pre_read_rows;
```
Comment (AshinGau) [inline be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:459]: We should remove the useless columns(filter columns) and clean the predicate columns.

**Comment by AshinGau** [comment]
LGTM

File: be/src/vec/exec/format/parquet/vparquet_group_reader.cpp (hunk 1 above)
Comment (kaka11chen) [reply be/src/vec/exec/format/parquet/vparquet_group_reader.cpp:459]: It has remove in above codes.

**Comment by kaka11chen** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.78 seconds
 stream load tsv:          424 seconds loaded 74807831229 Bytes, about 168 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230425070905_clickbench_pr_134625.html

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**